base/linalg/tridiag.jl

```diff
@@ -312,7 +321,19 @@ immutable Tridiagonal{T} <: AbstractMatrix{T}
     d::Vector{T}     # diagonal
     du::Vector{T}    # sup-diagonal
     du2::Vector{T}   # supsup-diagonal for pivoting
+
+    Tridiagonal(dl::Vector{T}, d::Vector{T}, du::Vector{T}, du2::Vector{T}) = new(dl, d, du, du2)
+    function Tridiagonal(n::Integer, m::Integer)
+        if n != m
+            throw(DimensionMismatch("Tridiagonal matrix must be square"))
+        end
+        new(Vector{T}(n-1), Vector{T}(n), Vector{T}(n-1), zeros(T, n-2))
```

Comment:
Yup, the usage of `du2` is anything but clear to me, so I figured initializing to zero is the safe thing to do. I think `du2` is only written to during factorization and hence does not need to be initialized, but if someone could confirm that, I'd appreciate it.